frontend/web/components/pages/WidgetPage.tsx

```diff
@@ -1,18 +1,30 @@
-import React, {Component, useEffect, useState} from 'react';
```

Comment:
This is a refactor of the Widget page to fix all the types, as part of this I have regression tested the integration. It fixes a few states 
![image](https://user-images.githubusercontent.com/8608314/221149407-c79834f8-b298-4348-bde9-6240cfcd5f6a.png)